### ISSUE TITLE
Log exception details when building and analyzing the `CSharpExplodedGraph`

### ISSUE THREAD

**Issue by andrei-epure-sonarsource** [SonarSource/sonar-dotnet#2542]
Currently in [FlowAnalysisExtensions ](https://github.com/SonarSource/sonar-dotnet/blob/7.15.0.8572/sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/FlowAnalysisExtensions.cs) we are not handling exceptions that are thrown in [CSharpControlFlowGraphBuilder](https://github.com/SonarSource/sonar-dotnet/blob/7.15.0.8572/sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/ControlFlowGraph/CSharpControlFlowGraphBuilder.cs)

These exceptions are caught by Roslyn and only the error message is shown in the analysis logs as a warning AD0001 such as : 

```
"C:\Workspace\external-projects\beyourmarket\src\BeYourMarket.Core\BeYourMarket.Core.csproj" (default target) (7:3) -> (CoreCompile target) ->
  CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.EmptyNullableValueAccess' threw an exception of type 'Syste m.NotSupportedException' with message 'BreakStatement'. [C:\Workspace\external-projects\beyourmarket\src\BeYourMarket. Core\BeYourMarket.Core.csproj]
```

Roslyn has open tickets to log more errors: dotnet/roslyn#24346 and dotnet/roslyn#1455

Unfortunately, this doesn't help our users nor us to find out what the problem is.

We should log the stacktrace (on a single line) and enable these failures to catch them during our QA steps.

SonarCsharp version 7.15

Related ticket: #2541
